src/components/API.js

```diff
@@ -0,0 +1,93 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+  _checkServerRes = (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  };
+
+  //   01. loading user info from the server
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._checkServerRes);
+  }
+
+  //   02. getInitial cards data
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._checkServerRes);
+  }
+
+  //   03. edit profile
+  editProfile(userData) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: userData.name,
```

Comment:
There is an error in the console when I try to update the profile `API.js:33 Uncaught TypeError: Cannot read properties of undefined (reading 'name')
    at Api.editProfile (API.js:33:24)
    at PopupWithForm._handleFormSubmit (index.js:181:8)
    at HTMLFormElement.<anonymous> (PopupWithForm.js:39:12)`